### PR TITLE
[WIP] First commit for tvm-conv2d benchmark code

### PR DESCRIPTION
**Pull request by Joejiong**
This is the first commit for tvm-conv2d benchmark code:
Modifications are made in the following two files:
- readme.md
- tvm-conv2d.py

### CODE REVIEW

File: examples/conv-opt/comparison/README.md [hunk 1]
```diff
@@ -33,6 +33,15 @@ $ cd buddy-mlir/examples/conv-opt/comparison/
 $ python3 pytorch-conv2d.py
 ```
 
+- TVM
+> **_NOTE:_**  In order to run tvm-conv2d.py, one has to manually build tvm with their system. For building reference please check TVM official installation guide: https://tvm.apache.org/docs/install/from_source.html.
+
+```
+$ cd buddy-mlir/examples/conv-opt/comparison/
+$ python3 tvm-conv2d.py 3
```
Comment (zhanghb97) [inline examples/conv-opt/comparison/README.md:41]:
```
$ python3 tvm-conv2d.py 3
usage: tvm-conv2d.py [-h] [--size N]
tvm-conv2d.py: error: unrecognized arguments: 3
```

It seems that the command should be:

```
python3 tvm-conv2d.py --size 3
```

**Review by zhanghb97 — CHANGES_REQUESTED** [review]
Good! But there are some details that should be fixed, and I have left comments.

File: examples/conv-opt/comparison/tvm-conv2d.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+import argparse
+
+import cv2
+import tvm
+from tvm import te
+import numpy as np
+
+IMAMGE_PATH = '../images/YuTu2048.png'
+
+TILED_HEIGHT, TILED_WEIGHT = 3, 3  # Tile sizes for height and weight
```
[the remark this answers is not in the document]
Comment (Joejiong) [reply examples/conv-opt/comparison/tvm-conv2d.py:25]: Actually the tile-size is taken from their(TVM) tutorial, but it really should be decided by brute-force-like auto-search module, here I manual tried some configurations find the default one is better.

File: examples/conv-opt/comparison/README.md (hunk 1 above)
Comment (Joejiong) [reply examples/conv-opt/comparison/README.md]: right, done

File: examples/conv-opt/comparison/tvm-conv2d.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    n, k = data.shape[-1], weight.shape[-1]
+
+    start = time.time()
+    mod(data, weight, out)
+    end = time.time()
+    print("evaluation time: %s" % ((end - start)))
+    out = out.asnumpy().squeeze()
+    return out
+
+
```
[the remark this answers is not in the document]
Comment (Joejiong) [reply examples/conv-opt/comparison/tvm-conv2d.py]: done

File: examples/conv-opt/comparison/tvm-conv2d.py [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+    parser = argparse.ArgumentParser(description='kernel size')
+    parser.add_argument('--size', metavar='N', type=int, default=3, help='an integer describe kernel size')
+    args = parser.parse_args()
+    
+    # test for optimized version 
+    # test_cache_optimization()
+
+    main(args.size)
+
+
```
[the remark this answers is not in the document]
Comment (Joejiong) [reply examples/conv-opt/comparison/tvm-conv2d.py]: done

File: examples/conv-opt/comparison/tvm-conv2d.py [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+                      [7, 6, 5, 4, 0, -4, -5, -6, -7], 
+                      [6, 5, 4, 3, 0, -3, -4, -5, -6], 
+                      [5, 4, 3, 2, 0, -2, -3, -4, -5], 
+                      [4, 3, 2, 1, 0, -1, -2, -3, -4]], dtype='float32')
+
+sobel_3x3_filter = sobel_3x3.reshape((1, 1, 3, 3))
+sobel_5x5_filter = sobel_5x5.reshape((1, 1, 5, 5))
+sobel_7x7_filter = sobel_7x7.reshape((1, 1, 7, 7))
+sobel_9x9_filter = sobel_9x9.reshape((1, 1, 9, 9))
+
```
[the remark this answers is not in the document]
Comment (Joejiong) [reply examples/conv-opt/comparison/tvm-conv2d.py:61]: done

File: examples/conv-opt/comparison/tvm-conv2d.py [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+import cv2
+import tvm
+from tvm import te
+import numpy as np
+
+IMAMGE_PATH = '../images/YuTu2048.png'
+
+TILED_HEIGHT, TILED_WEIGHT = 3, 3  # Tile sizes for height and weight
+
+TARGET = 'llvm -mcpu=skylake-avx512'
```
[the remark this answers is not in the document]
Comment (Joejiong) [reply examples/conv-opt/comparison/tvm-conv2d.py]: sure, done

File: examples/conv-opt/comparison/tvm-conv2d.py [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+    print(tvm.lower(sch, [X, K, Y], simple_mode=True))
+
+    start = time.time()
+    mod(data, weight, out)
+    end = time.time()
+    print("evaluation time: %s" % ((end - start)))
+    out = out.asnumpy().squeeze()
+    return out
+
+
```
[the remark this answers is not in the document]
Comment (Joejiong) [reply examples/conv-opt/comparison/tvm-conv2d.py]: done

**Review by zhanghb97 — APPROVED** [review]
Thanks! Remember to add a new line to the end of the file for the next time.

**Comment by Joejiong** [comment]
> Thanks! Remember to add a new line to the end of the file for the next time.

Thanks, ok

**Pull request merged**